### PR TITLE
Ticket648 add ioc cp2800

### PR DESCRIPTION
**Pull request by LiamPanchaud**

### CODE REVIEW

File: cp2800Sup/cp2800.db [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    field(ONVL, "1")
+    field(TWVL, "2")
+    field(THVL, "3")
+    field(FRVL, "4")
+    field(FVVL, "5")
+    field(SXVL, "6")
+    field(ZRST, "Illegal RSP")
+    field(ZRSV, "MAJOR")
+    field(ONST, "OK")
+    field(TWST, "Invallid Cmd")
```
Comment (Alistair-McGann-Tessella) [inline cp2800Sup/cp2800.db]: Nit: spelling error

File: cp2800Sup/cp2800.db [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+    field(LLSV, "MAJOR")
+    field(HSV,  "MINOR")
+    field(ADEL, "5")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:ELAPSED")
+    field(SDIS, "$(P)DISABLE")
+}
+
+
+######### FOR FUTURE USE
```
Comment (Alistair-McGann-Tessella) [inline cp2800Sup/cp2800.db]: Now there is a commit with these records for future use in, they no longer need to be on this branch as comments. They can be deleted safely, and can be refered to later at this commit

File: cp2800Sup/cp2800.proto [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+# TEMP_1    = 'c' 0x0D 0x8F 1;         # Output water temperature (0.1 degC)
+# TEMP_2    = 'c' 0x0D 0x8F 2;         # Helium temperature (0.1 degC)
+# TEMP_3    = 'c' 0x0D 0x8F 3;         # Oil temperature (0.1 degC)
+# H_ALP     = 'c' 0xBB 0x94 0;         # Average low side pressure (0.1 PSIA)
+# H_AHP     = 'c' 0x7E 0x90 0;         # Average high side pressure (0.1 PSIA)
+
+
+
+
+#test { out "%s"; in "%40c"; }
```
Comment (Alistair-McGann-Tessella) [inline cp2800Sup/cp2800.proto]: Are these still used?

File: cp2800Sup/cp2800.proto [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+comp_min { 
+    out $CP_CMD $COMP_MIN "%01<sum8>%1!"; 
+    in "%1!" $CP_RSP $COMP_MIN "%4r%01<sum8>"; 
+    wait $DLY; 
+}
+
+
+# temp_0   { out $CP_CMD $TEMP_0   "%01<sum8>%1!"; in "%1!" $CP_RSP $TEMP_0   "%4r%01<sum8>"; wait $DLY; }
```
Comment (Alistair-McGann-Tessella) [inline cp2800Sup/cp2800.proto]: Are these protocols  parked to be used in the future too? If this is the case, I would recommend deleting them here and revisiting this commit when they are needed.
If they aren't needed for the future, these should definitely be deleted

File: cp2800Sup/cp2800.db [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+    field(ZNAM, "Enable")
+    field(ONAM, "Disable")
+    field(OSV,  "MAJOR")
+    field(VAL,  "0")
+    field(PINI, "YES")
+}
+
+### CMDRSP 
+
+record(mbbiDirect, "$(P)CMDRSP")
```
Comment (Alistair-McGann-Tessella) [inline cp2800Sup/cp2800.db]: Some of your PV names are very short, I feel this makes them less intuitive to read. Is it possible to expand on these names? Particularly CMDRSP, CLRRSP and RSP.

File: cp2800Sup/cp2800.db [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+}
+
+### COMMANDS 
+
+record(stringin,"$(P)ID") 
+{
+    field(DESC, "Product ID")
+    field(DTYP, "stream")
+    field(INP,  "@cp2800.proto prod_id($(P)) $(PORT)")
+    field(SCAN, "10 second")
```
Comment (Alistair-McGann-Tessella) [inline cp2800Sup/cp2800.db]: Do we expect the ID of the device to change? If not, can this field be removed

File: cp2800Sup/cp2800.proto [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+# TEMP_0    = 'c' 0x0D 0x8F 0;         # Input water temperature (0.1 degC)
+# TEMP_1    = 'c' 0x0D 0x8F 1;         # Output water temperature (0.1 degC)
+# TEMP_2    = 'c' 0x0D 0x8F 2;         # Helium temperature (0.1 degC)
+# TEMP_3    = 'c' 0x0D 0x8F 3;         # Oil temperature (0.1 degC)
+# H_ALP     = 'c' 0xBB 0x94 0;         # Average low side pressure (0.1 PSIA)
+# H_AHP     = 'c' 0x7E 0x90 0;         # Average high side pressure (0.1 PSIA)
+
+
+
```
Comment (Alistair-McGann-Tessella) [inline cp2800Sup/cp2800.proto:43]: These protocols are very complicated. Some rough documentation about what they do any why would be useful to have in the protocol file.
More fleshed out documentation should be on the wiki

**Review by Alistair-McGann-Tessella — CHANGES_REQUESTED** [review]

File: cp2800Sup/cp2800.proto (hunk 7 above)
Comment (LilithCole) [reply cp2800Sup/cp2800.proto:43]: Unfortunately this code wasn't written by us, and no documentation for it is aviable.

Comment (kjwoodsISIS) [reply cp2800Sup/cp2800.proto:43]: The fundamental lesson of the HIFI Cryomag project is that undocumented code & systems come back to bite us in a big way.  Do we really want to be adding undocumented IOCs to our code base?  
Can we track down the original author?  Judging from  Liam's initial comments this IOC appears to have come from the MICE project.  The MICE project is (for now) based at ISIS, so the author might still be available to us.

Comment (LiamPanchaud) [reply cp2800Sup/cp2800.proto:43]: I'm reasonably confident that there is a document in the manuals folder (I can't remember the name) which documents how the CP2800 comms protocol works. It's essentially a dictionary of variables that can be referenced with hash keys. To read from one, it's prepended with 'c', to write I _think_ it's prepended with 'a'. 

e.g. reading high side pressure is 'c' to read, `0x7E 0x90` which is the hash key for the high side pressure member of the  dictionary, then followed by a `0`, but I can't remember what that does anymore.

Comment (Alistair-McGann-Tessella) [reply cp2800Sup/cp2800.proto:43]: Thanks Liam and Chris, I'm now happy that the wiki page and the manual on the share document the comms protocol. I've added a note in the protocol file as a signpost for these.

**Review by Alistair-McGann-Tessella — APPROVED** [review]

**Pull request merged**